src/components/canrawview/canrawview.cpp

```diff
@@ -76,6 +81,41 @@ QJsonObject CanRawView::getConfig() const
     return config;
 }
 
+void CanRawView::setConfig(const QObject& qobject)
+{
+    Q_D(CanRawView);
+
+    for (const auto& p: getSupportedProperties())
+    {
+        QVariant v = qobject.property(p.first.toStdString().c_str());
+        if (!v.isValid() || v.type() != p.second.first)
+            continue;
+
+        d->_props[p.first] = v;
+    }
+}
+
+std::shared_ptr<QObject> CanRawView::getQConfig() const
+{
```

Comment:
Duplicated code? Can be make that general, and in one place?

Reply:
For getQConfig yes, it actually is general, although I didn't see a proper place to put it. The only common place is ComponentInterface - and I wasn't sure if that would be ok - what do you think about it?
As for setConfig(const QObject&) I don't think it is possible, as changing certain parameters will probably require some reconfiguration (as e.g. backend and interface in Device)

Reply:
It would be great to have new class/type to mix-in these features for reuse (not an OOP "interface", but a simple container of functionalities). It is a common part, so it can go into the `common` directory. If possible, can be provided with a separate commit.